### ISSUE TITLE
plexus-form up for adoption

### ISSUE THREAD

**Issue by odf** [AppliedMathematicsANU/plexus-form#50]
As I have mentioned several times before, I cannot really spare the time to maintain this code as a proper project, so as of now, I am officially declaring it abandoned. I may or may not be able to pick things up at some point in the future, but would then almost certainly do a complete rewrite from scratch.

If anyone has a fork or otherwise related project that they are planning to maintain for some time and would like people to know about, please comment below and I will add a link to the readme. Also feel free to use this thread for coordinating future efforts.

Since my employer holds the copyright to the code, I cannot pass this repository over to a new maintainer, and I will also have to hold on to the name plexus-form on npm.

Cheers,
Olaf

**Comment by ismaelga** [comment]
Hello,

I don't mind to keep my [fork](https://github.com/ismaelga/plexus-form) active by accepting PR and fix some issues.

So, if no one else volunteers to fully maintain it, you can point out to my fork on the readme.

btw, thanks for the work you have done here, and to AppliedMathematicsANU.

**Comment by bebraw** [comment]
@ismaelga Cool. :+1:

Maybe it would be the best if we could get the npm version point at that? If that's not possible, then you would have to pick a different name to publish it under.

**Comment by ismaelga** [comment]
It's up to you. You could add me as a package contributor on npmjs.com so I can release new versions of it. If not, I can create my own package.

**Comment by bebraw** [comment]
@ismaelga One alternative might be to see if we could get the project under [rackt](https://github.com/rackt). I can ask around to see if there's interest.

**Comment by ismaelga** [comment]
Sure! That would be awesome :+1:

If it helps, you can tell them I'll contribute to its development/maintenance.

**Comment by bebraw** [comment]
@ismaelga Ok, I opened the discussion. We'll see how that goes.

CC @ryanflorence

**Comment by bebraw** [comment]
@ismaelga No dice so far... Perhaps the best idea would be to fork the project, rename, and start developing. Let me know if you need help with some bits. I've done my share of work with plexus-form.

**Comment by odf** [comment]
I think a fork with a new name would be best. I have a number of small packages with the plexus- prefix, which doesn't really mean anything outside of our internal project here at ANU which they were created for.

If at some point I get around to doing more active work on form generation for React apps, I'll be happy to contribute back to the forked project.

**Comment by ismaelga** [comment]
Should I also fork the plexus-* dependencies? Or you will keep maintaining them?

**Comment by odf** [comment]
I wouldn't mind maintaining those as long as it's just a matter of adding bug fixes and small tweaks here and there, which seems likely. But there's probably an argument to be made for just pulling them into the plexus-forms code base. They're each just a single, relative small source file right now.

The idea for having plexus-validate separately was that people could plug in their own validator, but then they can still potentially do that even if it's included. It would make more sense for it to be a separate package if it were meant to be a complete JSON-Schema validator, but that was never my intention. If you're interested in extending plexus-validate in that direction, it might be best to fork it.

As for plexus-objective, that's really just a collection of small utility functions to make mangling objects and arrays a bit more pleasant. Even though I use them all over the place, I'm now finding it easier to copy and paste the ones I need between code bases rather than synchronizing with a separate repo.

So long story short, it's up to you. If you're not planning on changing much, keep using the existing projects, otherwise fork them or pull them in, depending on your preferences.

**Comment by ismaelga** [comment]
I think I will go with pulling them to main package then.

Hopefully, next week I will have some time to work on the fork, and release it as a new package. Will come back to you when it feels ready to me.

**Comment by datapimp** [comment]
@ismaelga how is this coming? can i help? i'd love to find a well maintained json-schema and react forms solution

**Comment by ismaelga** [comment]
@datapimp Sadly I haven't done anything yet. Will do my best to get things done on the weekend.
Sure you can help! For starters, finding a new name for the project :smile: Also, I can give you contributor permissions to the repo if you want.

**Comment by ismaelga** [comment]
Finally got time for this. I decided that *react-json-editor* is a good name and here it is https://github.com/ismaelga/react-json-editor

@odf will you add it to plexus-form README or you want me to send a PR for that?

**Comment by odf** [comment]
@ismaelga Many thanks for stepping up! I've added the link now. If you want to change the wording, though, I'm happy to accept your PR.
